coding-the-legislation/40_legislation_evolutions.md

```diff
@@ -60,67 +61,71 @@ we get the output:
-        ...
-
-class progressive_income_tax(Variable):
-    column = FloatCol
-    entity = Person
-    label = u"Former tax replaced by the flat tax on the 1st of Jan 2014"
-    stop_date = date(2013, 12, 31)
-    definition_period = MONTH
-
-    def function(person, period, legislation):
+    def formula(person, period, legislation):
         ...
 ```
 
+will return a value calculated by its `formula` function if it is called for any date before its `end = '2014-12-31'` date. 
+While it will return the default OpenFisca value for `FloatCol` if it is called after its `end` date.
+
 Note that:
-- A variable can of course have both a `start_date` and a `stop_date`.
-- The `stop_date` is the last day a formula is valid, and not the first day it is not valid anymore.
+- The `end` is the last day a variable and its formulas are valid (not the first day of unvalidity).
+- The `end` value is a string of 'YYYY-MM-DD' format where YYYY, MM and DD are respectively a year, month and day.
 - When defining a date, the month is given **before** the day.
 
 ## Formula evolving over time
 
 Some fiscal or benefit mechanism significantly evolve over time, with bigger changes than a simple parameter adjustement.
 
-For instance, let's assume that from the 1st of Jan. 2017, our previous `flat_tax_on_salary` will not apply on the first `1000` earned by the person. We can implement this rule with a `DatedVariable`:
+For instance, let's assume that from the 1st of Jan. 2017, our previous `flat_tax_on_salary` will not apply on the first `1000` earned by the person. 
+We can implement this rule with dated formulas:
 
 ```py
-class flat_tax_on_salary(DatedVariable):
+class flat_tax_on_salary(Variable):
     column = FloatCol
     entity = Person
     label = u"Individualized and monthly paid tax on salaries"
     definition_period = MONTH
 
-    @dated_function(start = date(2017, 1, 1))
-    def function_2017(person, period, legislation):
+    def formula_2017(self, simulation, period):
         salary = person('salary', period)
         salary_above_1000 = min_(salary - 1000, 0)
         return salary_above_1000 * legislation(period).taxes.salary.rate
 
-    @dated_function(start = date(2014, 01, 01), stop = date(2016, 12, 31))
-    def function_2014(person, period, legislation):
+    def formula_2014(self, simulation, period):
+        salary = person('salary', period)
+        salary_above_1000 = min_(salary - 1000, 0)
+        return salary_above_750 * legislation(period).taxes.salary.rate
+
+    def formula(self, simulation, period):
         salary = person('salary', period)
 
         return salary * legislation(period).taxes.salary.rate
 ```
 
-
 Note that:
-- If you omit the start date, the formula is valid until the stop date.
-- If you omit the stop date, it is valid from the start date.
+- A formula name should start with `formula`.
+- To define a starting date on a formula, we add a suffix to its name: `formula_2017` is valid from '2017'.
```

Comment:
If we go to this level of detail, also write that the suffix is made of an underscore followed by a date.